addons/mail/static/src/discuss/web/thread_service/thread_service_patch.js

```diff
@@ -0,0 +1,18 @@
+/* @odoo-module */
+
+import { ThreadService } from "@mail/core/common/thread_service";
+
+import { patch } from "@web/core/utils/patch";
+
+patch(ThreadService.prototype, {
+    _openChatWindow(
+        thread,
+        replaceNewMessageChatWindow,
+        { autofocus = true, openMessagingMenuOnClose } = {}
+    ) {
+        if (thread.type === "channel" && !thread.is_pinned) {
+            this.store.env.services["bus_service"].addChannel(thread.busChannel);
+        }
+        super._openChatWindow(...arguments);
+    },
+});
```

Comment:
Why not `bus_service.addChannel` on every `thread.open()` of a `discuss.channel`?

`_openChatWindow` is a very specific function, would be easier to keep functional semantics when this applies on a more functionally significant entry-point, which `thread.open()` is.

Also why limit only on `discuss.channel` of type `channel` and when the thread is unpinned? I'd say all `discuss.channel` support real-time messaging so they can all `bus_service.addChannel()`.

Reply:
Yeah can get rid of checking thread `type === "channel"` but IMO we don't need to do `addChannel` again and again for every thread we open, that's why only for the threads which are `unpinned` , also we're doing this only on `_openChatWindow` cause it works perfectly fine in other cases

Reply:
This makes the code harder to maintain, in the sense that this looks quite complicated, to the point that we'd be scared to adapt because the conditional and specific code being overridden looks like something that needs a lot of deep understanding with the current code, therefore refactoring it becomes a huge effort.

This is a `[FIX]` and if we can do an equivalent functional fix with much fewer LOCs and code maintenance cost, I would strongly go with this equivalent shorter code.

Having all opening of `discuss.channel` enabling the real-time aspect of messaging is easier to adapt than all opening of chat window of channels of type channel that are not pinned should have it enabled. Besides `_openChatWindow`

Reply:
Okay, I'll add it to `thread.open()` then, without any conditional check

Reply:
As I rebased and tested it again, we no longer need to do `addChannel(thread.busChannel)`, so the issue is sorted.